File: materials/2019-2020/11a/practice/2020-02-26-unit-tests/1va_zad/src/main/java/com/elsys/Rectangle.java
package main.java.com.elsys;

import static java.lang.StrictMath.abs;

public class Rectangle {
    Point A;
    Point B;
    double a;
    double b;

    public Rectangle(Point A, Point B){
        this.A=A;
        this.B=B;
        a=abs(A.x-B.x);
        b=abs(A.y-B.y);
    }

    public double getPerimeter(){
        return (double) ((a+b)*2);
    }
    public double getArea(){
        return (double) (a*b);
    }
}

File: materials/2021-2022/11a/2022-01-05-abstract/rectangle.cpp
#include "rectangle.hh"
#include <iostream>

using namespace std;

rectangle::rectangle(Point ul, Point lr) {
  this->ul = ul;
  this->lr = lr;
}

void rectangle::print() const {
  cout << "I am rectangle at: " << ul.to_string() << " and " << lr.to_string()
       << endl;
}